connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -340,13 +342,19 @@ public void startTask(ConnectorTaskId id,
         // Start the task before adding modifying any state, any exceptions are caught higher up the
         // call chain and there's no cleanup to do here
         workerTask.initialize(taskConfig);
+        WorkerTask existing = tasks.put(id, workerTask);
+        if (existing != null) {
+            String msg = "Task already exists in this worker; the herder should not have requested "
+                    + "that this : " + id;
```

Comment:
Copied log message without reading it from the start of the method. I'll fix both. 
By the way, this check kind of needs to happen twice because multi-threaded execution is allowed. First to avoid constructing a new task if there is one with the same id already, and at the end, to make sure that indeed there's no such task in `tasks` map by the time the thread is adding it. 

On a side note, I would have used putIfAbsent above. But Java 7 doesn't define this method for Map interface (was added in Java 8). Thus, I would have to drop the use of `Map` for the member field, and declare it as concrete type `ConcurrentHashMap`. Or we don't care about this overwriting and we leave things as is. We should pick the lesser evil tradeoff in this case.